### DISCUSSION TITLE
ModuleNameAttribute issue on new Module

### DISCUSSION THREAD

**Discussion by johndcollins** [OrchardCMS/OrchardCore#11119] [Q&A] [answered]
I'm trying to make a my own module for a project. My module is not picked up the the OrchardCore framework because the ModuleNameAttribute is not in my .net core assembly. 

I've created the manifest in my module, but I'm not sure how to get my module to add this custom attribute to my web assembly.

When I look at the sample OrchardCore.Cms.Web I see the attributes of the standard modules, but I don't see my custom module.
![Screenshot 2022-02-03 092657](https://user-images.githubusercontent.com/1684130/152396515-7ce2883c-92f2-4a4d-9de6-63ae0deea4f7.png)

**Comment by ns8482e** [comment, marked as answer]
Add reference to OrchardCore.Module.Targets https://www.nuget.org/packages/OrchardCore.Module.Targets/

**Reply by johndcollins** [reply]
Thanks for your reply. 
I've created my own Module.Targets as I'm not using the OrchardCms, I'm trying to do my own version. My Module.Target is similar to the Orchard one,  with some namespace changes.

I was assuming this was the key here:
`
    <ItemGroup>
      <AssemblyAttribute Include="OrchardCore.Modules.Manifest.ModuleMarkerAttribute">
        <_Parameter1>$(AssemblyName)</_Parameter1>
        <_Parameter2>$(ModuleType)</_Parameter2>
      </AssemblyAttribute>
      <AssemblyAttribute Include="OrchardCore.Modules.Manifest.ModuleAssetAttribute">
        <_Parameter1>%(ModuleAssets.Identity)</_Parameter1>
      </AssemblyAttribute>
    </ItemGroup>
`

If I add the OrchardCore.Module.Targets and remove mine I get a .Net SDK Error:
"Duplicate 'EmbeddedResource' items were included. The .NET SDK includes 'EmbeddedResource' items from your project directory by default...."

**Reply by ns8482e** [reply]
OC does lot more than this - you need to use `OrchardCore.Module.Targets` for your module to work for OC

**Reply by johndcollins** [reply]
I think I was editing my reply when you replied, sorry.

If I add the OrchardCore.Module.Targets and remove mine I get a .Net SDK Error:
"Duplicate 'EmbeddedResource' items were included. The .NET SDK includes 'EmbeddedResource' items from your project directory by default...."

**Reply by ns8482e** [reply]
Remove any `EmbeddedResource` entry from your csproj - OC Module.Targets does it for you

**Reply by johndcollins** [reply]
I'm not sure I understand, sorry. 

This is a copy of the OrchardCore.Admin module just changed the name. This is just for testing. Why would I want to remove the Views and wwwroot folder as it would have my specific resources.

![Screenshot 2022-02-03 095338](https://user-images.githubusercontent.com/1684130/152401006-e34486fd-9891-4fee-9c28-fc26d1e4f0ad.png)

**Reply by ns8482e** [reply]
I mean not remove from the project - remove any explicit `<EmbeddedResource />` entry from `.csproj` file using edit project file - that caused you error `Duplicate 'EmbeddedResource'`

**Reply by johndcollins** [reply]
Ok thanks, I got it cleaned up and compiling. 
I added the OrchardCore.Module.Target to all my modules. I made a custom Application.Cms.Target and Application.Cms.Core.Target and added my modules to these. I included this in my Web.Cms project and compiled and l'm still not seeing the ModuleName attributes on my assembly.
![Screenshot 2022-02-03 105012](https://user-images.githubusercontent.com/1684130/152409182-bfaf35d5-ceba-4d74-b624-a13acb6ea610.png)

**Reply by johndcollins** [reply]
Ok, I think I got it. I had to also add the OrchardCore.Application.Targets to my Web Project.
Thanks for your help.
